src/coreclr/jit/fginline.cpp

```diff
@@ -653,12 +653,12 @@ class SubstitutePlaceholdersAndDevirtualizeWalker : public GenTreeVisitor<Substi
                         }
 
                         call->GetSingleInlineCandidateInfo()->exactContextHandle = context;
-#ifdef DEBUG
-                        // Update inline context for Debug.
-                        call->gtInlineContext = call->GetSingleInlineCandidateInfo()->inlinersContext;
-                        DebugInfo debugInfo(call->gtInlineContext, ILLocation(call->gtRawILOffset, false, true));
+
+                        // Update inline context and DebugInfo for the new inline candidate.
+                        InlineContext* inlineContext = call->GetSingleInlineCandidateInfo()->inlinersContext;
+                        INDEBUG(call->gtInlineContext = inlineContext);
+                        DebugInfo debugInfo(inlineContext, inlineContext->GetLocation());
                         m_curStmt->SetDebugInfo(debugInfo);
```

Comment:
What is the idea behind updating the debug information here? The regular inliner does not update the debug information.

Have you verified that this fixes the failures in superpmi-asmdiffs-checked-release?

Reply:
I see that `impMarkInlineCandidate` ends up using `compInlineContext`. It does not look like it is being set correctly when it is called above.

Reply:
I'm puzzled. `impMarkInlineCandidate` set `InlinersContext` using `compInlineContext`, which is supposed to be the right context, so what we should do here is exactly to update the `call->gtInlineContext` with the new `inlineContext`. But CI isn't happy...
I'm updating the `DebugInfo` just to address yet another assertion where it complains about the `InlineContext` is mismatched.

Reply:
`compInlineContext` does not seem like the right context to me. Nothing is setting it above... It should be set based on information that was only available during the import that created the call being devirtualized here.

Reply:
That sounds good to me.